### PR TITLE
Fix datadog-agent image builder

### PR DESCRIPTION
**Pull request by ostankin**
[INF-2070](https://salemove.atlassian.net/browse/INF-2070)

Merged together [Jenkinsfile](https://github.com/salemove/alertlogic-agent/blob/master/Jenkinsfile) for alertlogic-agent and the existing [build job](https://ci.at.samo.io/job/build_sm-docker-dd-agent/configure) that has been decommissioned.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -12,6 +12,8 @@ withResultReporting(slackChannel: '#tm-inf') {
     stage('Build') {
       checkout(scm)
       shortCommit = sh(returnStdout: true, script: 'git log -n 1 --pretty=format:"%h"').trim()
+      ddVersion = sh(returnStdout: true, script: 'grep FROM Dockerfile|cut -d ":" -f2').trim()
+      imageTag = "${ddVersion}-${shortCommit}"
```
Comment (deiwin) [inline Jenkinsfile]: `shortCommit` is defined above, but doesn't have to anymore, when it's only used in the scope of this `stage`. I'd change `def image, shortCommit` -> `def image, imageTag`, `shortCommit = sh` -> `def shortCommit = sh`, and `ddVersion = sh` -> `def ddVersion = sh`.

File: Jenkinsfile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 @Library('cve-monitor') __
 
 def MAIN_BRANCH                    = 'master'
-def DOCKER_REPOSITORY_NAME         = 'alertlogic-agent'
+def DOCKER_REPOSITORY_NAME         = 'dd-agent'
 def DOCKER_REGISTRY_URL            = 'https://662491802882.dkr.ecr.us-east-1.amazonaws.com'
 def DOCKER_REGISTRY_CREDENTIALS_ID = 'ecr:us-east-1:ecr-docker-push'
```
Comment (deiwin) [inline Jenkinsfile]: > Note that now images will be stored in ECR, so `image` in the daemonset must be updated accordingly.

Why move it to ECR?

Comment (ostankin) [reply Jenkinsfile]: As I understand our approach to storing Docker images, everything must to be stored in ECR unless explicitly needs to be public (e.g. browser images). Do we need to keep our `dd-agent` public?

File: Jenkinsfile (hunk 1 above)
Comment (ostankin) [reply Jenkinsfile]: Agree, this makes sense. Fixed: f5c0107

**Pull request merged**